pkg/sbproxy/notifications/producer.go

```diff
@@ -283,9 +285,15 @@ func (p *Producer) ping(ctx context.Context, done chan<- struct{}) {
 }
 
 func (p *Producer) connect(ctx context.Context) error {
+	connectContext, correlationID, err := createConnectContext(ctx)
+	if err == nil {
```

Comment:
don't you want to stop here already if err != nil ?

Reply:
no I want to proceed as it today

Reply:
i didn't understand it in first look, maybe add a comment?